src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ExportClient/OtlpExportClient.cs

```diff
@@ -83,14 +90,26 @@ protected HttpRequestMessage CreateHttpRequest(byte[] buffer, int contentLength)
             request.Headers.Add(header.Key, header.Value);
         }
 
-        // TODO: Support compression.
+        Stream data = new MemoryStream(buffer, 0, contentLength);
```

Comment:
The change from `ByteArrayContent` to `MemoryStream` + `StreamContent` introduces a performance regression for the non-compressed path.
When compression is **not** enabled (which is the default), we're adding an extra allocation per export (2 objects vs. 1 previously). This is a hot path executed on every export batch.
Users who haven't opted into compression should not pay this cost.